src/imagery/i.eodag/i.eodag.py

```diff
@@ -139,54 +151,91 @@ def get_bb(vector=None):
+    if not vector:
+        return get_bb()
+
+    # TODO: Read Vector WKB
+
+
 def download_by_id(query_id: str):
-    gs.verbose(
-        _(
-            "Searching for product ending with: {}".format(
-                query_id[-min(len(query_id), 8) :]
-            )
-        )
-    )
-    product, count = dag.search(id=query_id)
+    gs.message(_("Attempting to download product: {}".format(query_id)))
+    product, count = dag.search(id=query_id, provider=options["provider"])
     if count != 1:
-        raise ParameterError("Product couldn't be uniquely identified")
+        raise ParameterError("Product couldn't be uniquely identified.")
     if not product[0].properties["id"].startswith(query_id):
-        raise ParameterError("Product wasn't found")
-    gs.verbose(
-        _("Poduct ending with: {} is found.".format(query_id[-min(len(query_id), 8) :]))
-    )
-    gs.verbose(_("Downloading..."))
+        raise ParameterError("Product wasn't found.")
+    gs.verbose(_("Poduct {} is found.".format(query_id)))
+    gs.verbose(_("Downloading {}".format(query_id)))
     dag.download(product[0])
 
 
+def ids_from_file_txt(ids_file_txt):
+    ids_set = set()
+    with open(ids_file_txt, "r") as ids_stream:
+        line_index = 0
+        lines = ids_stream.read().split("\n")
+        for line in lines:
+            line_index += 1
+            if not line:
+                continue
+            line = line.strip()
+            if line.find(" ") != -1:
+                gs.warning(
+                    _(
+                        'File "{}", line {}, has space(s). Skipping line... '.format(
+                            ids_file_txt, line_index
+                        )
+                    )
+                )
+                continue
+            ids_set.add(line.strip())
+    return ids_set
+
+
 def download_by_ids(products_ids):
     # Search in all recognized providers
     for product_id in products_ids:
         try:
             download_by_id(product_id)
-        except ParameterError:
-            gs.error(
-                _(
-                    "Product ending with: {}, failed to download".format(
-                        product_id[-min(len(id), 8) :]
-                    )
-                )
-            )
+        except ParameterError as e:
+            gs.error(e)
+            gs.error(_("Product {} failed to download".format(product_id)))
+
+
+def parse_id_option(id_option_string):
+    to_download_ids = id_option_string.split(",")
+    ids_set = set()
+    for to_download_id in to_download_ids:
+        ids_set.add(to_download_id.strip())
+    return ids_set
+
+
+def parse_file_option(file_option_string):
+    ids_set = set()
+    files_list = file_option_string.split(",")
+    for file in files_list:
+        try:
+            ids_set.update(ids_from_file_txt(file))
+        except FileNotFoundError:
+            gs.warning(_('Couldn\'t read file "{}", Skipping file...'.format(file)))
+    return ids_set
```

Comment:
Is there a use-case for multiple file input?

In principle I think the ids and file option could be merged. if the string input to _ids_ is an existing file path, you could read the ids from that file, if not you treat the input as a (list of) id(s).... No?

Reply:
I agree with @ninsbl, if the user has multiple text files with ids, they can always merge them before hand.